crates/wysiwyg/src/composer_model/format.rs

```diff
@@ -51,7 +53,63 @@ where
 
     pub fn inline_code(&mut self) -> ComposerUpdate<S> {
         self.push_state_to_history();
-        self.format_or_unformat(InlineFormatType::InlineCode)
+        let format_type = InlineFormatType::InlineCode;
+        if self.action_is_reversed(format_type.action()) {
+            self.unformat(format_type)
+        } else {
+            self.add_inline_code()
+        }
+    }
+
+    /// Finds the closest structure node ancestor for each leaf node handle and groups it with other
+    /// leaves that share it as the common closest structure node ancestor. If none is found,
+    /// the root/document node is used instead.
+    pub(crate) fn group_leaves_by_closest_structure_ancestors(
+        &self,
+        leaves: Vec<&DomLocation>,
+    ) -> HashMap<DomHandle, Vec<DomLocation>> {
+        let mut structure_ancestors = HashMap::new();
+        for leaf in leaves {
+            let first_structure_ancestor =
+                self.find_structure_ancestor(&leaf.node_handle);
+            // Get the closest ancestor path or the root one (empty Vec) if there is none
+            let ancestor_handle = first_structure_ancestor
+                .map(|a| a.clone())
+                .unwrap_or(DomHandle::root());
+            let list = structure_ancestors
+                .entry(ancestor_handle)
+                .or_insert(Vec::new());
+            // Add the DomHandle of the leaf to the list of grouped handles by this ancestor
+            list.push(leaf.clone());
+        }
+        structure_ancestors
+    }
+
+    pub(crate) fn remove_and_clean_up_empty_nodes_until(
```

Comment:
I think you should re-use something like delete_nodes in crates/wysiwyg/src/composer_model/delete_text.rs here. It handles some corner cases I think.

Reply:
Yup, same as below, I had a feeling that we already had something like this but I couldn't really find it. I'll see if it works here too.

Reply:
Actually, it seems like it doesn't. A parent structure node can be temporarily empty and this method would actually remove it too, while previously we'd just stop at it if we found it.

Reply:
OK, no need to re-use it then but maybe a comment about why it is different.